### PR TITLE
Display: Do not offer a tty if it's already taken by a logind session

### LINKED ISSUE #1588: User switching breaks
We should look into whether we are switching ttys properly...

https://bugs.kde.org/show_bug.cgi?id=458169

### PR DESCRIPTION
**Pull request by aleixpol**
Otherwise, it all freezes and nothing works as it should.

Fixes #1588

### CODE REVIEW

File: src/daemon/Display.cpp [hunk 1]
```diff
@@ -50,7 +54,30 @@
...
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    int fetchAvailableVt() {
+        const auto vt = VirtualTerminal::currentVt();
+        return approveTty(QStringLiteral("tty%1").arg(vt)) ? vt : VirtualTerminal::setUpNewVt();
```
Comment (davidedmundson) [inline src/daemon/Display.cpp]: On BSD:

currentVt will be -1
approveTTY will return true as we have no logind

so we return -1.  Same for if currentVt fails on Linux.

We want:
```
const auto vt = VirtualTerminal::currentVt();
if (vt > 0) {
    if (approveVt(vt) {
        return vt;
    }
} 
return setupNewVt();
```

File: src/daemon/Display.cpp [hunk 2]
```diff
@@ -50,7 +54,30 @@
 #include "WaylandDisplayServer.h"
 
 
-namespace SDDM {
+namespace SDDM
+{
+    bool approveTty(const QString &desiredTty) {
```
Comment (davidedmundson) [inline src/daemon/Display.cpp]: I don't like the name. "approve"  sounds like a verb, implying it does something active, also it's not a term logind or ioctl's are using

bool isTtyInUse(...)

File: src/daemon/Display.cpp (hunk 1 above)
Comment (aleixpol) [reply src/daemon/Display.cpp]: good catch, I wonder why it worked on the testing then... :thinking:

**Comment by plfiorini** [comment]
Wonder what happens on systems without logind

**Review by davidedmundson — APPROVED** [review]

**Review by plfiorini — APPROVED** [review]

**Comment by arrowd** [comment]
On FreeBSD we use SDDM with consolekit, which probably has its own way to find out if the vt is free.

**Comment by plfiorini** [comment]
Yeah, for now `isTtyInUse()` returns false but in the future it can do whatever is needed for consolekit

**Pull request merged**